interface-definitions/system_syslog.xml.in

```diff
@@ -98,6 +98,14 @@
                   <valueless/>
                 </properties>
               </leafNode>
+              <leafNode name="local-host-name">
```

Comment:
I tend to vouch for `host-name` here, We have a similar option for DHCP: `set interfaces ethernet eth1 dhcp-options host-name`

This would also suffice for a XML building block that can be re-used in those two places.

Reply:
Could you make it a suggestion that we could commit then?

Reply:
I don't think we can reuse the building block since in all other cases, the host name is the name only, while here it can be the fqdn.

Reply:
I see where you are heading now. So the main issue is that the `preserve-fqdn` CLI option does not behave as expected.

Why not fix this implicitly?
If `preserve-fqdn` is defined on the CLI, then read on `system host-name` and `system-domain-name` from the configuration into the dictionary responsible to rsyslog.

Then automatically render the content for the `$LocalHostName` variable. This would save us adding one more nerd knob on the CLI.

Reply:
Makes sense, unless someone comes up with a use case for setting a completely different host-name we could avoid adding another config option. I'll update the PR accordingly.

Reply:
Updated PR and re-run smoke tests